test/extensions/filters/http/ext_proc/filter_test.cc

```diff
@@ -96,6 +96,16 @@ class HttpFilterTest : public testing::Test {
             async_client_stream_info_.upstreamInfo());
     EXPECT_CALL(testing::Const(*mock_upstream_info), upstreamHost());
 
+    // Let the dispatcher_.time_system_ pointing to a MockTimeSystem object.
+    mock_time_system_ = new testing::NiceMock<MockTimeSystem>();
```

Comment:
please try SimulatedTimeSystem. I had attempted to deprecate MockTimeSystem though there were are few difficult-to-remove cases. THe doc for MockTimeSystem has a TODO to me to get rid of it :)

PIng me if you have questions but I think for the most part things should be easier. You will control advancement of time from the test thread.

Reply:
Sure, done!

After changing to use SimulatedTimeSystem, we are calling

test_time_->timeSystem().setMonotonicTime(MonotonicTime(std::chrono::microseconds(x)));

before each filter_callbacks_->dispatcher().timeSource().monotonicTime() call to get the right latency:

https://github.com/envoyproxy/envoy/blob/76667dd9454101a6f9d38b672555e076706d190a/source/extensions/filters/http/ext_proc/processor_state.cc#L31

https://github.com/envoyproxy/envoy/blob/76667dd9454101a6f9d38b672555e076706d190a/source/extensions/filters/http/ext_proc/processor_state.cc#L41